include/okapi/impl/control/async/asyncMotionProfileControllerBuilder.hpp

```diff
@@ -0,0 +1,135 @@
+/**
+ * @author Ryan Benasutti, WPI
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#pragma once
+
+#include "okapi/api/chassis/controller/chassisController.hpp"
+#include "okapi/api/control/async/asyncLinearMotionProfileController.hpp"
+#include "okapi/api/control/async/asyncMotionProfileController.hpp"
+#include "okapi/api/util/logging.hpp"
+#include "okapi/impl/device/motor/motor.hpp"
+#include "okapi/impl/device/motor/motorGroup.hpp"
+#include "okapi/impl/util/timeUtilFactory.hpp"
+
+namespace okapi {
+class AsyncMotionProfileControllerBuilder {
+  public:
+  /**
+   * A builder that creates async motion profile controllers. Use this to build an
+   * AsyncMotionProfileController or an AsyncLinearMotionProfileController.
+   */
+  AsyncMotionProfileControllerBuilder();
+
+  /**
+   * Sets the output.
+   *
+   * @param ioutput The output.
+   * @return An ongoing builder.
+   */
+  AsyncMotionProfileControllerBuilder &withOutput(const Motor &ioutput,
+                                                  const QLength &idiameter,
+                                                  const AbstractMotor::GearsetRatioPair &ipair);
```

Comment:
Document the other params (here and below). Also document that this should be used for an ALMPC and the other versions should be used for an AMPC.